js/ux-capture.js

```diff
@@ -22,91 +33,114 @@ window.UX = (function() {
   let onMark;
   let onMeasure;
 
-  return {
-    expect: function(zones) {
-      expectedZones = zones.map(zone => {
-        // only create promises if zone contains any marks, otherwise just ignore it
-        if (zone.marks && zone.marks.length > 0) {
-          // wrap each string in the array into an object so we can attach record() methods to them
-          zone.marks = zone.marks.map(label => ({ label }));
-
-          // create a promise for each expected mark
-          const promises = zone.marks.map(
-            mark =>
-              new Promise((resolve, reject) => {
-                mark.record = () => {
-                  if (isUserTimingSupported) {
-                    // record the mark using W3C User Timing API
-                    performance.mark(mark.label);
-
-                    // if callback is specified, call it with mark label
-                    if (onMark) {
-                      onMark(mark.label);
-                    }
-                  }
-
-                  /**
-                   * Report same mark on Chrome/Firefox timeline
-                   *
-                   * keep in mind, these timestams are counted from timeline recording start
-                   * while UserTiming marks are counted from navigationStart event
-                   * however visually, they all will be offset by the same amount of time and align vertically on the charts
-                   *
-                   * (we'd provide a helper to highlight discrepancy, but unfortunately,
-                   * there is no way to know when in timeline did navigationStart event occured)
-                   */
-                  if (isConsoleTimeStampSupported) {
-                    console.timeStamp(mark.label);
+  window.UX.expect = function(zones) {
+    expectedZones = zones.map(zone => {
```

Comment:
I know this is old code, but do we want to prevent `expect` from being called multiple times in some way, or warn if it's called more than once to avoid accidental overwrites of `expectedZones`? Alternatively, should we just `concat` with existing `expectedZones` on subsequent calls?

Reply:
No, it's not old code, just first iteration. There are a couple features (#7, #8 and those related to interactive views) that might require a decision on our part if we allow calling `UX.expect()` multiple times.

I'd defer final logic of overriding/append/block till we have that functionality flushed out and leave it as is for now.

Reply:
Is it worth adding a comment here to remind us to reevaluate this logic?